services/worker-manager/src/providers/aws.js

```diff
@@ -196,8 +196,8 @@ class AwsProvider extends Provider {
     }
 
     const {document, signature} = workerIdentityProof;
-    if (!document || !signature) {
-      throw new ApiError('Token validation error');
+    if (!document || !signature || !(typeof document === "string")) {
+      throw new ApiError('Request must include both a document (string) and a signature');
```

Comment:
I thought we wanted to keep the error messages vague on purpose? In case of a malicious action

Reply:
The vagueness was about the validity of the document -- this is just about adherence to the expected form of the request, which is public information anyway.